src/Sarif.Multitool.Library/OptionsInterpretter.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+using Microsoft.CodeAnalysis.Sarif.Driver;
+
+namespace Microsoft.CodeAnalysis.Sarif.Multitool
+{
+    public class OptionsInterpretter
+    {
+        //  Poor man's dependency injection
+        public OptionsInterpretter() : this (new EnvironmentVariableGetter())
+        {
+
+        }
+        
+        public OptionsInterpretter(IEnvironmentVariableGetter environmentVariableGetter)
+        {
+            if (environmentVariableGetter == null)
+            {
+                throw new ArgumentException("Required parameter", nameof(environmentVariableGetter));
```

Comment:
>ArgumentException [](start = 26, length = 17)

Change to argumentNullException. With that, u can you just the nameof(xyz)

Reply:
you could also do this: _environmentVariableGetter = environmentVariableGetter ??  throw new ArgumentNullException(nameof(enviromentVariableGetter));

---
In reply to: [566070993](https://github.com/microsoft/sarif-sdk/pull/2273#discussion_r566070993) [](ancestors = 566070993)

Reply:
Done.

---
In reply to: [566071879](https://github.com/microsoft/sarif-sdk/pull/2273#discussion_r566071879) [](ancestors = 566071879,566070993)